File: backend/src/issues/issue.model.ts
import mongoose from 'mongoose';

const issueSchema = new mongoose.Schema({
    username: { type: String, required: true, unique: true },
    bookboxId: { type: String, required: true },
    subject: { type: String, required: true },
    description: { type: String, required: true },
    status: { type: String, enum: ['open', 'in_progress', 'resolved'], default: 'open' },
    reportedAt: { type: Date, default: Date.now },
    resolvedAt: { type: Date, default: null },
});

// Add indexes for faster querying
issueSchema.index({ username: 1, bookboxId: 1 }, { unique: true });
issueSchema.index({ status: 1 });
issueSchema.index({ reportedAt: -1 }); 

// Delete resolved issues after 7 days
issueSchema.index({ resolvedAt: 1 }, { expireAfterSeconds: 604800 });

const Issue = mongoose.model('Issue', issueSchema, "issues");

export default Issue;

File: app/lib/models/issue_model.dart
class Issue {
  final String id;
  final String username; 
  final String bookboxId;
  final String subject;
  final String description;
  final String status; 
  final DateTime reportedAt;
  final DateTime? resolvedAt;

  Issue({
    required this.id,
    required this.username,
    required this.bookboxId,
    required this.subject,
    required this.description,
    required this.status,
    required this.reportedAt,
    this.resolvedAt,
  });

  factory Issue.fromJson(Map<String, dynamic> json) {
    return Issue(
      id: json['id'],
      username: json['username'],
      bookboxId: json['bookboxId'],
      subject: json['subject'],
      description: json['description'],
      status: json['status'],
      reportedAt: DateTime.parse(json['reportedAt']),
      resolvedAt: json['resolvedAt'] != null ? DateTime.parse(json['resolvedAt']) : null,
    );
  }
}